datajunction-server/datajunction_server/api/notifications.py

```diff
@@ -121,3 +124,19 @@ async def get_user_notification_preferences(
         for pref in notification_preferences
     ]
     return JSONResponse(content=response)
+
+
+@router.get("/notifications/users")
+async def get_users_for_notification(
+    entity_name: str,
+    entity_type: EntityType,
+    session: AsyncSession = Depends(get_session),
+) -> JSONResponse:
+    """Get users for the given notification preference"""
+    notification_preferences = await get_entity_notification_preferences(
+        session=session,
+        entity_name=entity_name,
+        entity_type=entity_type,
+    )
+    users = [perf.user.username for perf in notification_preferences]
+    return JSONResponse(content=users)
```

Comment:
Can these have typed responses rather than just `JSONResponse`? So in this case, something like:
```
@router.get("/notifications/users")
async def get_users_for_notification(
    entity_name: str,
    entity_type: EntityType,
    session: AsyncSession = Depends(get_session),
) -> list[str]:
     ...
```

Reply:
+1